### PR TITLE
Fix: using git token_id on Templates always assumes GitLab

### LINKED ISSUE #545: Using git `token_id` on Templates always assumes GitLab
Git tokens in env0 can be used to integrate with "Other VCS", GitLab, and soon Azure DevOps.  
Setting `token_id` on this provider [always assumes](https://github.com/env0/terraform-provider-env0/blob/75af2bebbc1b57400acfa1d6d68e54c577b7fd86/env0/resource_template.go#L342) we're using Gitlab, causing the request to fail.  

# Steps to reproduce:
Apply:  
```terraform
resource "env0_environment" "foobar" { #could also be just a template_resource
  name        = "foobar"
  project_id  = data.env0_project.default_project.id
  
  without_template_settings {
    repository = "https://github.com/env0/templates"
    token_id = "68ba48c9-ec6e-4177-a130-e6f3ffb02f9b"
  }
}
```  

This will fail with:  
```
│ Error: could not update template: 400 Bad Request: You have no access to that repository
```  

Examining the request made by the client shows the template is marked as **gitlab** despite not setting `isGitlab: true`

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #545 

### Solution

1. Added is_gitlab to the schema and updated the code.
2. Updated the API struct.
3. Updated the acceptance tests.
4. As part of an effort to replace the linter, I'm slowly doing other small fixes.

### CODE REVIEW

File: .golangci.yaml
Comment (TomerHeber) [inline .golangci.yaml:1]: this is for a new linter... ignore unrelated changes... it's a lot of work so I'm slowly adding it to the PRS.

File: env0/resource_template.go [hunk 1]
```diff
@@ -385,14 +391,10 @@ func templateCreatePayloadFromParameters(prefix string, d *schema.ResourceData)
 
 	isNew := d.IsNewResource()
 
-	tokenIdKey := "token_id"
```
Comment (TomerHeber) [inline env0/resource_template.go:388]: not required anymore. Gitlab should be explicitly configured. This is a breaking change that was discussed in Slack.
Need to add to release notes.

File: env0/resource_template_test.go [hunk 2]
```diff
@@ -628,7 +633,7 @@ func TestUnitTemplateResource(t *testing.T) {
 				Description:          templateUseCase.template.Description,
 				GithubInstallationId: templateUseCase.template.GithubInstallationId,
 				IsGitlabEnterprise:   templateUseCase.template.IsGitlabEnterprise,
-				IsGitLab:             templateUseCase.template.TokenId != "" && !templateUseCase.template.IsAzureDevOps,
```
Comment (TomerHeber) [inline env0/resource_template_test.go:631]: now just check the field excplicitly.

File: env0/resource_environment.go
Comment (Wassap124) [inline env0/resource_environment.go:1]: does this have anything to do with the issue?

File: env0/resource_variable_set.go
Comment (Wassap124) [inline env0/resource_variable_set.go:1]: does this have anything to do with the issue?

File: env0/resource_variable_set_assignment.go
Comment (Wassap124) [inline env0/resource_variable_set_assignment.go:1]: does this have anything to do with the issue?

File: env0/data_project.go
Comment (Wassap124) [inline env0/data_project.go:1]: does this have anything to do with the issue?

**Review by Wassap124 — APPROVED** [review]
well, it seems that you indeed do what's requested,
but i have a lot of questions

**Comment by Wassap124** [comment]
[Can't create template using provider and git token](https://trello.com/c/KAWaWBBW/13651-cant-create-template-using-provider-and-git-token)
